File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/adapter/inbound/zeebe/newsletter/SendWelcomeMailWorker.java
package io.miragon.example.adapter.inbound.zeebe.newsletter;

import io.camunda.client.annotation.JobWorker;
import io.camunda.client.annotation.Variable;
import io.miragon.example.adapter.process.generated.NewsletterSubscriptionProcessApi.TaskTypes;
import io.miragon.example.application.port.inbound.newsletter.SendWelcomeMailUseCase;
import io.miragon.example.domain.SubscriptionId;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class SendWelcomeMailWorker {
	
	private static final Logger log = LoggerFactory.getLogger(SendWelcomeMailWorker.class);
	
	private final SendWelcomeMailUseCase useCase;
	
	public SendWelcomeMailWorker(SendWelcomeMailUseCase useCase) {
		this.useCase = useCase;
	}
	
	@JobWorker(type = TaskTypes.NEWSLETTER_SEND_WELCOME_MAIL)
	public void handle(@Variable String subscriptionId) {
		log.debug("Received job to send welcome mail for subscriptionId: {}", subscriptionId);
		useCase.sendWelcomeMail(new SubscriptionId(UUID.fromString(subscriptionId)));
	}
	
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/adapter/inbound/zeebe/newsletter/SendWelcomeMailWorker.kt
package io.miragon.example.adapter.inbound.zeebe.newsletter

import io.miragon.example.adapter.process.generated.NewsletterSubscriptionProcessApi.TaskTypes
import io.miragon.example.application.port.inbound.newsletter.SendWelcomeMailUseCase
import io.miragon.example.domain.SubscriptionId
import io.camunda.client.annotation.JobWorker
import io.camunda.client.annotation.Variable
import mu.KotlinLogging
import org.springframework.stereotype.Component
import java.util.*

@Component
class SendWelcomeMailWorker(
    private val useCase: SendWelcomeMailUseCase
) {

    private val log = KotlinLogging.logger {}

    @JobWorker(type = TaskTypes.ACTIVITY_SEND_WELCOME_MAIL)
    fun handle(@Variable subscriptionId: String) {
        log.debug { "Received job to send welcome mail for subscriptionId: $subscriptionId" }
        useCase.sendWelcomeMail(SubscriptionId(UUID.fromString(subscriptionId)))
    }
}
